nessie/jobs/refresh_canvas_data_catalog.py

```diff
@@ -139,3 +140,27 @@ def generate_external_catalog(self, external_schema, schema_df):
 
         # For debugging process, export to external_table_ddl to file to get a well formed SQL template for canvas-data
         return external_table_ddl
+
+    # Gets an inventory of all the tables by tracking the S3 canvas-data daily location and run count verification to ensure migration was successful
+    def verify_external_data_catalog(self):
+        s3_client = s3.get_client()
+        bucket = app.config['LOCH_S3_BUCKET']
+        external_schema = app.config['REDSHIFT_SCHEMA_CANVAS']
+        prefix = get_s3_canvas_daily_path(datetime.now() - timedelta(days=1))
+        app.logger.info(f'Daily path = {prefix}')
+        directory_names = []
+        s3_objects = s3_client.list_objects_v2(Bucket=bucket, Prefix=prefix)
+        for object_summary in s3_objects['Contents']:
+            # parse table names from the S3 object URLs
+            directory_names.append(object_summary['Key'].split('/')[3:4][0])
```

Comment:
@sandeepmjay Any reason not to write this as `object_summary['Key'].split('/')[3]`?

Reply:
Yep, makes sense. Looks like a hangover from my scratchpad. Switched it :)